watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Users/EditUserFragment.java

```diff
@@ -104,35 +112,37 @@ private void initializeViews() {
 
         mName.setText(mUser.getName());
         mEmail.setText(mUser.getEmail());
+    }
 
-        mSubmit = (Button) mParentActivity.findViewById(R.id.profile_edit_submit);
-        mSubmit.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                Utility.hideKeyboard(mParentActivity, mLayout);
-                boolean hasErrors = false;
-                if (!mPassword.getText().toString().equals(mReenterPassword.getText().toString())) {
-                    mReenterPassword.setError("Your passwords don't match!");
-                    hasErrors = true;
-                }
-
-                if (mName.getText().toString().isEmpty()) {
-                   mName.setError("Name can't be blank!");
-                   hasErrors = true;
-                }
-
-                Pattern regex = Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);
-                Matcher matcher = regex.matcher(mEmail.getText().toString());
-                if (!matcher.find()) {
-                    mName.setError("Email must be valid!");
-                    hasErrors = true;
-                }
-
-                if (hasErrors) return;
-
-                sendEditUserRequest();
-            }
-        });
+    private void validateUser() {
+        Utility.hideKeyboard(mParentActivity, mLayout);
+        boolean hasErrors = false;
+
+        if (mConfirm.getText().toString().length() == 0) {
+            mConfirm.setError("You need to enter in your current password before saving!");
+            hasErrors = true;
+        }
+
+        if (!mPassword.getText().toString().equals(mReenterPassword.getText().toString())) {
+            mReenterPassword.setError("Your passwords don't match!");
+            hasErrors = true;
+        }
+
+        if (mName.getText().toString().isEmpty()) {
+            mName.setError("Name can't be blank!");
+            hasErrors = true;
+        }
+
+        Pattern regex = Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);
+        Matcher matcher = regex.matcher(mEmail.getText().toString());
+        if (!matcher.find()) {
+            mName.setError("Email must be valid!");
+            hasErrors = true;
+        }
+
+        if (hasErrors) return;
+
+        sendEditUserRequest();
     }
 
     /**
```

Comment:
There's a debugging log statement here <code> Log.i("saved things", "lol");<\code>. We should remove that.